### PR TITLE
#174797534 Add Search Functionality

### PR DESCRIPTION
**Pull request by florisng**
### What does this PR do?
Allow users to search for requests

### Description of Task to be completed?

- Pull the repo to get the project's latest version
- Create a branch to work on
- Rebase
- Create an endpoint that takes a given search in params
- Create controller for searching requests
- Run `npm run dev`
- Write Tests
- Add, commit and push changes

### How should this be manually tested?

1. Clone the repo `git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
2. Switch your working directory `cd script-heroes-bn-backend`
3. Checkout the branch git checkout `ft-search-functionality-#174797534`
4. Install all dependencies `npm install`
5. Run tests `npm run dev`
6. Login on `/api/auth/login`
7. Make sure you have some trips already created in DB
8. Search Trips by: 
     - Start date `/api/allTripsByStartDate/search?startDate=date`
     - End date `/api/allTripsByEndDate/search?endDate=date`
     - Origin `/api/allTripsByOrigin/search?origin=countryFrom`
     - Destination `/api/allTripsByDest/search?destination=countryTo`
     - Both Origin & Destination `/api/allTripsByOriginDest/search?origin=countryFrom&destination=countryTo`
9. Search Requests by: 
     - Status `/api/allRequestsByStatus/search?status=status` (status = pending/approved/rejected)
     - id `/api/getRequestById/search?id=requestId`
     - user id `/api/getRequestByUser/search?id=userId`
10. Also run `npm run test`

Screenshot (93)

### What are the relevant pivotal tracker stories?

[#174797534](https://www.pivotaltracker.com/n/projects/2465784/stories/174797534)

### CODE REVIEW

**Review by dushimeemma — APPROVED** [review]
Please work on tests to increase coverage and also work on writting PR description otherwise good job

File: src/controllers/forgotPassword.js [hunk 1]
```diff
@@ -28,10 +28,8 @@ export default {
           )
         });
       })
-      .catch(() => {
-        res.status(400).send({
-          err: res.__('Failed in sending Reset password email !!!')
-        });
+      .catch((err) => {
+        res.status(400).send({ err });
```
Comment (Adadie) [inline src/controllers/forgotPassword.js:32]: I think it would be better if all responses were in JSON format

File: src/controllers/profile.js [hunk 2]
```diff
@@ -19,15 +19,15 @@ class ProfileController {
       return res
         .status(400)
         .json({ error: res.__('Profile is already complete') });
-    } else {
```
Comment (Adadie) [inline src/controllers/profile.js:22]: Why are we removing else?

File: src/database/seeders/20201015053129-User.js [hunk 3]
```diff
@@ -4,61 +4,62 @@ module.exports = {
...
+        language: 'English',
+        currency: 'USD',
+        country: 'Rwanda',
+        department: 'Marketing',
+        linemanager: '2',
+        roleId: '2',
+        createdAt: new Date(),
+        updatedAt: new Date()
+      },
+      {
```
Comment (Adadie) [inline src/database/seeders/20201015053129-User.js:26]: May you not change the indentation

File: test/index.test.js [hunk 4]
```diff
@@ -20,16 +20,4 @@ describe('landing page', () => {
...
-      .get('/notFound')
-      .end((err, res) => {
-        if (err) done(err);
-        res.should.have.status(404);
-        res.body.should.be.a('object');
-        res.body.should.have.property('errors');
-        done();
-      });
-  });
 });
```
Comment (Adadie) [inline test/index.test.js:23]: I think removing this test will reduce the coverage for that file

File: src/controllers/profile.js (hunk 2 above)
Comment (florisng) [reply src/controllers/profile.js:22]: `else` is unnecessary after `return`

**Review by ericus123 — APPROVED** [review]
Good

**Comment by kipropbrian** [comment]
Do not merge!

**Pull request merged**